src/java.desktop/macosx/classes/sun/lwawt/macosx/CPrinterJob.java

```diff
@@ -610,25 +636,37 @@ private boolean jobSetup(Pageable doc) {
 
     // The following methods are CPrinterJob specific.
 
-    @Override
-    @SuppressWarnings("removal")
-    protected void finalize() {
-        synchronized (fNSPrintInfoLock) {
-            if (fNSPrintInfo != -1) {
-                dispose(fNSPrintInfo);
+    static class NSPrintInfoDisposer implements DisposerRecord {
+
+        private long fNSPrintInfo;
+        private Object lock;
+
+        NSPrintInfoDisposer(Object l, long ptr) {
+            lock = l;
+            fNSPrintInfo = ptr;
+        }
+
+
+        public void dispose() {
+            synchronized (lock) {
+                if (fNSPrintInfo != -1) {
+                    CPrinterJob.disposeNSPrintInfo(fNSPrintInfo);
+                }
+                fNSPrintInfo = -1;
             }
-            fNSPrintInfo = -1;
         }
     }
 
-    private native long createNSPrintInfo();
-    private native void dispose(long printInfo);
+    private static native long createNSPrintInfo();
+    private static native void disposeNSPrintInfo(long printInfo);
```

Comment:
What is the reason for using static here when it was not before?

Reply:
because it has no need for the instance and moreover it can't be called on the instance because that would mean it is still reachable.
It is called instead from the disposer.